LayoutTests/fast/scrolling/mac/scrollable-area-size-for-overlay-scrollbars.html

```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ AsyncOverflowScrollingEnabled=true MockScrollbarsEnabled=false ] -->
+<html>
+<head>
+    <style>
+        .container {
+            height: 200px;
+            width: 200px;
+            border: 1px solid black;
+            overflow: scroll;
+        }
+        .contents {
+            height: 500px;
+            width: 500px;
+            background-color: silver;
+        }
+    </style>
+    <script src="../../../resources/js-test.js"></script>
+    <script src="../../../resources/ui-helper.js"></script>
+    <script>
+        if (window.testRunner)
+            testRunner.waitUntilDone();
+        description("Tests that scrollable area size is properly updated when toggling scrollbar style");
+        window.internals.setUsesOverlayScrollbars(true);
+
+        var scroller;
+        async function doTest()
+        {
+            preChangeSize = window.internals.scrollableAreaSize(document.getElementById("scroller"));
+            window.internals.setUsesOverlayScrollbars(false);
+            await UIHelper.ensurePresentationUpdate();
+            shouldBeGreaterThan("preChangeSize", "window.internals.scrollableAreaSize(document.getElementById(\"scroller\"))");
```

Comment:
I think you could do this without new Internals API. You could just have a `width=100%` element inside the scrollbar and get its `offsetWidth`.

Reply:
So the original test actually didn't properly expose the bug as the prior code did properly change the size of the scrollable area. The only way I found to expose the bug was via getting at the actual contents size of the scrollable area, which I think we can only do via the internals api.